### PR TITLE
Added setutils.[]= variant

### PR DESCRIPTION
**Pull request by beef331**
Adds a variant of setutils.[]= which takes a whole set

### CODE REVIEW

File: lib/std/setutils.nim [hunk 1]
```diff
@@ -75,3 +75,19 @@ func `[]=`*[T](t: var set[T], key: T, val: bool) {.inline.} =
     s[a3] = true
     assert s == {a2, a3}
   if val: t.incl key else: t.excl key
+
+func `[]=`*[T](s: var set[T], args: set[T], val: bool) {.inline.} =
+  ## Syntax sugar for `if val: t.incl args else: t.excl args`,
```
Comment (konsumlamm) [inline lib/std/setutils.nim]:
```suggestion
  ## Syntax sugar for `if val: s.incl args else: s.excl args`,
```

**Review by konsumlamm — CHANGES_REQUESTED** [review]

**Review by timotheecour — APPROVED** [review]
LGTM

(in this or future PR, we should also change:
```nim
func `[]=`*[T](t: var set[T], key: T, val: bool) {.inline.} =
```

to 
```nim
func `[]=`*[T](s: var set[T], key: T, val: bool) {.inline.} =
```

**Comment by Araq** [comment]
Do other languages support it? It does look a bit weird to be honest.

**Comment by timotheecour** [comment]
> Do other languages support it? It does look a bit weird to be honest.

`set` support in other languages is rather limited though (closest thing is usually a notion of bit vector). Plausible options are:

put ?
putAll ?
[]= ?

(but keeping []= for the previous overload from https://github.com/nim-lang/Nim/pull/17272)

**Comment by Araq** [comment]
Strong -1 from me, I cannot imagine a use case for this, seems like pure bloat. "Consistency" isn't a use case.

**Comment by Araq** [comment]
Sorry, it simply looks too weird and it's unclear we need it.

**Pull request closed without merging**